packages/create-react-app/index.js

```diff
@@ -173,10 +173,14 @@ function install(packageToInstall, verbose, callback) {
 function run(root, appName, version, verbose, originalDirectory, template) {
   var packageToInstall = getInstallPackage(version);
   var packageName = getPackageName(packageToInstall);
-
   install(packageToInstall, verbose, function(code, command, args) {
     if (code !== 0) {
-      console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
+      console.log();
+      console.error('Aborting installation.', chalk.cyan(command + ' ' + args.join(' ')), 'has failed.');
+      console.log();
+      console.log('Deleting', chalk.cyan(appName), 'from', chalk.cyan(originalDirectory));
+      fs.removeSync(path.join(originalDirectory, appName));
```

Comment:
You're right. Some use case could be a script that creates the project folder, initialises git then runs `create-react-app`. All files had the same birthtime.

So far I've only seen 4 possible files/directory in the project folder when installation fails

* package.json
* npm-debug.log
* yarn-error.log
* node_modules 

We could check if any of this files exist, delete them, check if project folder is empty then remove it?